pkg/addonmanager/addonclients/upgrader.go

```diff
@@ -51,6 +51,10 @@ func FluxChangeDiff(currentSpec, newSpec *cluster.Spec) *types.ChangeDiff {
 		logger.V(1).Info("Skipping Flux upgrades, not a self-managed cluster")
 		return nil
 	}
+	if currentSpec.Cluster.Spec.GitOpsRef == nil && newSpec.Cluster.Spec.GitOpsRef != nil {
+		logger.V(1).Info("Skipping Flux upgrades, already installed latest")
```

Comment:
should we add a comment here the explains why the diff does not take into account the addition of a 'new' flux version? I would find it a little confusing to be new to this code and see that the diff did not take into account the difference between no version and an existing version

```suggestion
		logger.V(1).Info("Skipping Flux upgrades, already installed latest version")
```

Reply:
i updated to
```
logger.V(1).Info("Skipping Flux upgrades, no previous flux installed in the cluster")
```
hopefully it's less confusing